client/src/common/components/Activities/components/ActivitiesList.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import Activity from './Activity';
+import Preloader from 'common/components/Preloader';
+import { fetchActivities } from '../model/actions';
+import { getActivities } from '../model/selectors';
+import MessageBox from 'common/components/MessageBox';
+import { MessageType } from 'common/constants/enums';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+
+class ActivitiesList extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    pending: false
+  };
+
+  componentDidMount() {
+    const { fetchActivities } = this.props;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(fetchActivities());
+    this.pendingPromise.promise
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pending: false });
+        }
+      });
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  render() {
+    const { activities } = this.props;
+    const { pending } = this.state;
+
+    return (
+      <div className="activities-list">
+        <h2 className="activities-list__heading">Activities</h2>
+        <div className="activities-list__content">
+          {activities.length ? (
```

Comment:
wouldn't be `activities ?` enough?

Reply:
Unfortunately no. This is an array of activities, so we have to check this array's length because if there are no activities, an empty array is returned. We check items array in the same way (ItemsList.jsx, line:45) but we can also check _empty(activities).